common/app/model/dotcomrendering/pageElements/PageElement.scala

```diff
@@ -186,12 +200,16 @@ object PageElement {
       case Text => for {
         block <- element.textTypeData.toList
         text <- block.html.toList
-        element <- Cleaner.split(text)
+        element <- HTML.tags(text)
       } yield { element match {
-        case ("h2", heading) => SubheadingBlockElement(heading)
-        case ("blockquote", blockquote) => BlockquoteBlockElement(blockquote)
-        case (_ , para) if (addAffiliateLinks) => AffiliateLinksCleaner.replaceLinksInElement(para, pageUrl = pageUrl, contentType = "article")
-        case (_, para) => TextBlockElement(para)
+        case ("h2", heading) =>
+          SubheadingBlockElement(heading)
+        case ("blockquote", blockquote) =>
+          BlockquoteBlockElement(blockquote)
+        case (_, para) =>
+          val tagLinks = Cleaners.tagLinks(article.tags, article.content.showInRelated, edition) _
+          val transforms = if (addAffiliateLinks) Cleaners.affiliateLinks(pageUrl) _ andThen tagLinks else tagLinks
```

Comment:
Might be slightly clearer if `Cleaners.affiliateLinks(pageUrl) _ ` was pulled out so you have `affiliateLinkCleaner andThen tagLinkCleaner else tagLinkCleaner` or similar.

Reply:
and/or if the affiliate link cleaner took `addAffiliateLinks` as an argument so you don't have any condition here it's just always `affiliateLinks andThen tagLinks`